hydra-node/test/Hydra/API/RestServerSpec.hs

```diff
@@ -2,29 +2,56 @@
 
 module Hydra.API.RestServerSpec where
 
-import Hydra.Prelude
+import Hydra.Prelude hiding (get)
 import Test.Hydra.Prelude
 
+import Data.Aeson (encode)
 import Data.Aeson.Lens (key)
 import Hydra.API.RestServer (DraftCommitTxRequest, DraftCommitTxResponse)
+import Hydra.API.Server (httpApp)
+import Hydra.API.ServerSpec (dummyChainHandle)
+import Hydra.Chain.Direct.Fixture (defaultPParams)
 import Hydra.Chain.Direct.State ()
 import Hydra.JSONSchema (prop_validateJSONSchema)
+import Hydra.Logging (nullTracer)
 import Test.Aeson.GenericSpecs (roundtripAndGoldenSpecs)
+import Test.Hspec.Wai (MatchBody (..), ResponseMatcher (matchBody), get, shouldRespondWith, with)
 import Test.QuickCheck.Property (property, withMaxSuccess)
 
 spec :: Spec
-spec = parallel $ do
-  roundtripAndGoldenSpecs (Proxy @(ReasonablySized DraftCommitTxResponse))
-  roundtripAndGoldenSpecs (Proxy @(ReasonablySized DraftCommitTxRequest))
-
-  prop "Validate /commit publish api schema" $
-    property $
-      withMaxSuccess 1 $
-        prop_validateJSONSchema @DraftCommitTxRequest "api.json" $
-          key "components" . key "messages" . key "DraftCommitTxRequest" . key "payload"
-
-  prop "Validate /commit subscribe api schema" $
-    property $
-      withMaxSuccess 1 $
-        prop_validateJSONSchema @DraftCommitTxResponse "api.json" $
-          key "components" . key "messages" . key "DraftCommitTxResponse" . key "payload"
+spec = do
+  parallel $ do
+    roundtripAndGoldenSpecs (Proxy @(ReasonablySized DraftCommitTxResponse))
+    roundtripAndGoldenSpecs (Proxy @(ReasonablySized DraftCommitTxRequest))
+
+    prop "Validate /commit publish api schema" $
+      property $
+        withMaxSuccess 1 $
+          prop_validateJSONSchema @DraftCommitTxRequest "api.json" $
+            key "components" . key "messages" . key "DraftCommitTxRequest" . key "payload"
+
+    prop "Validate /commit subscribe api schema" $
+      property $
+        withMaxSuccess 1 $
+          prop_validateJSONSchema @DraftCommitTxResponse "api.json" $
+            key "components" . key "messages" . key "DraftCommitTxResponse" . key "payload"
+
+    apiServerSpec
+
+-- REVIEW: we should add more tests for other routes here (eg. /commit)
```

Comment:
Yes. But what do you want to get reviewed here when you already know what we should do? IMO you could make this a TODO and provide a rationale in the PR.. or just do it?

Reply:
I promoted it to a todo. I'll do it in a separate PR since `/commit` is a bit more involved then `/protocol-parameters`